File: trackPixel.js
const express = require('express');
const fs = require('fs');
const app = express();

// Route for tracking pixel
app.get('/track', (req, res) => {
  const email = req.query.email;
  console.log("TRACKING");
  
  // Log the email open
  console.log(`Email opened by: ${email} at ${new Date().toISOString()}`);

  // Return a 1x1 transparent GIF
  const pixel = fs.readFileSync('transparent.gif');
  res.writeHead(200, {
    'Content-Type': 'image/gif',
    'Content-Length': pixel.length
  });
  res.end(pixel, 'binary');
});

// Start the server
app.listen(3000, () => {
  console.log('Tracking server is running on port 3000');
});

File: trackPixel.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"os"
	"time"
)

func trackHandler(w http.ResponseWriter, r *http.Request) {
	email := r.URL.Query().Get("email")
	log.Printf("Email opened by: %s at %s\n", email, time.Now().Format(time.RFC3339))

	pixel, err := ioutil.ReadFile("transparent.gif")
	if err != nil {
		http.Error(w, "Error loading pixel", http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "image/gif")
	w.Header().Set("Content-Length", fmt.Sprintf("%d", len(pixel)))
	w.Write(pixel)
}

func main() {
	http.HandleFunc("/track", trackHandler)

	// Get the port from the environment variable
	port := os.Getenv("PORT")
	if port == "" {
		port = "10000" // Default to port 10000 if PORT is not set
	}

	log.Printf("Server starting on port %s...\n", port)
	log.Fatal(http.ListenAndServe("0.0.0.0:"+port, nil)) // Bind to 0.0.0.0 to listen on all interfaces
}
